include/circt/Dialect/HW/HWTypesImpl.td

```diff
@@ -193,6 +193,8 @@ def UnionTypeImpl : HWType<"Union"> {
     FieldInfo getFieldInfo(::mlir::StringRef fieldName);
 
     ::mlir::Type getFieldType(::mlir::StringRef fieldName);
+    std::optional<unsigned> getFieldIndex(mlir::StringRef fieldName);
```

Comment:
Please use a sized type.  uint32_t or uint64_t.  We don't want the supported union size to depend on architecture.